src/libs/actions/Report.ts

```diff
@@ -5050,6 +5086,260 @@ function moveIOUReportToPolicy(reportID: string, policyID: string) {
+    const transactionsOptimisticData: Record<string, Transaction> = {};
+    const transactionFailureData: Record<string, Transaction> = {};
+    reportTransactions.forEach((transaction) => {
+        transactionsOptimisticData[`${ONYXKEYS.COLLECTION.TRANSACTION}${transaction.transactionID}`] = {
+            ...transaction,
+            amount: -transaction.amount,
+            modifiedAmount: transaction.modifiedAmount ? -transaction.modifiedAmount : 0,
+        };
+
+        transactionFailureData[`${ONYXKEYS.COLLECTION.TRANSACTION}${transaction.transactionID}`] = transaction;
+    });
+
+    optimisticData.push({
+        onyxMethod: Onyx.METHOD.MERGE_COLLECTION,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION}`,
+        value: transactionsOptimisticData,
+    });
+    failureData.push({
+        onyxMethod: Onyx.METHOD.MERGE_COLLECTION,
+        key: `${ONYXKEYS.COLLECTION.TRANSACTION}`,
+        value: transactionFailureData,
+    });
+
+    // We need to move the report preview action from the DM to the workspace chat.
+    const parentReportActions = allReportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${iouReport.parentReportID}`];
+    const parentReportActionID = iouReport.parentReportActionID;
+    const reportPreview = iouReport?.parentReportID && parentReportActionID ? parentReportActions?.[parentReportActionID] : undefined;
+    const oldChatReportID = iouReport.chatReportID;
+
+    if (reportPreview?.reportActionID) {
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oldChatReportID}`,
+            value: {[reportPreview.reportActionID]: null},
+        });
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${oldChatReportID}`,
+            value: {[reportPreview.reportActionID]: reportPreview},
+        });
+
+        // Add the reportPreview action to workspace chat
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticPolicyExpenseChatReportID}`,
+            value: {[reportPreview.reportActionID]: {...reportPreview, created: DateUtils.getDBTime()}},
+        });
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticPolicyExpenseChatReportID}`,
+            value: {[reportPreview.reportActionID]: null},
+        });
+    }
+
+    // Create the CHANGE_POLICY report action and add it to the expense report which indicates to the user where the report has been moved
+    const changePolicyReportAction = buildOptimisticChangePolicyReportAction(iouReport.policyID, policyID, true);
+    optimisticData.push({
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticPolicyExpenseChatCreatedReportActionID}`,
```

Comment:
This looks incorrect. We key reportActions by reportID, not reportActionID. This should just be the main reportID
```suggestion
        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`,
```

Reply:
@luacmartins This is not right, if we pass `reportID` looks like once the submitter is invited to the workspace, the report preview does not appear in the submitter's ws chat so IOU is not attached to anything:

https://github.com/user-attachments/assets/ce04fc3a-dd7d-4677-98f0-474ec8a38911

##

I noticed that we can completely remove `optimisticData / successData / failureData` from under `const changePolicyReportAction` and everything works as expected.

So we either leave `optimisticPolicyExpenseChatCreatedReportActionID` which you mentioned it looks incorrect or not pass the `optimisticData / successData / failureData` at all ? 🤔

Reply:
Hmm that's weird. Why would change the report on the `CHANGEPOLICY` action affect the preview?

Reply:
> Probably because the server is sending this data which makes it show up correctly.

@luacmartins Then you think I'd be ok to simply remove the `optimisticData / successData / failureData` or is there another reason for which you need that data sent to BE ?

And if there is a reason: what do you think we should pass given that passing `reportID` disappears the report preview in submitter's WS chat ?